### PR TITLE
[NIFI-12665] fetch parameter provider parameters

### PR DESCRIPTION
**Pull request by rfellows**
[NIFI-12665](https://issues.apache.org/jira/browse/NIFI-12665)

* add permissions checks to parameter provider table actions for edit, delete, and fetch.
* error handling
* routing to fetch parameter provider dialog
* list the parameter groups
* parameter sensitivity selction/deselection
* show banner error if any affected/referencing components are not readable and writeable
* add indicators to parameters for changed, new, removed, ...
* refactored parameter-references component to the common area. leveraged it in the fetch dialog.
* validate the fetch form
* submit the fetch parameter provider parameters

### CODE REVIEW

**Comment by mcgilman** [comment]
Reviewing....

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/parameter-providers/parameter-providers.reducer.ts [hunk 1]
```diff
@@ -59,11 +66,9 @@ export const parameterProvidersReducer = createReducer(
         status: 'success' as const
     })),
 
-    on(parameterProvidersApiError, (state: ParameterProvidersState, { error }) => ({
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/parameter-providers/parameter-providers.reducer.ts:62]: `error: null` above in `loadParameterProvidersSuccess` reducer can be removed.

**Comment by rfellows** [comment]
recheck

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/parameter-providers/fetch-parameter-provider-parameters/fetch-parameter-provider-parameters.component.ts [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+                        if (Object.hasOwn(this.parameterContextsToCreate, groupConfig.groupName)) {
+                            this.parameterContextsToCreate[groupConfig.groupName] = name;
+                        }
+                    });
+            });
+        }
+    }
+
+    submitForm() {
+        console.log('Form data', this.fetchParametersForm);
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/parameter-providers/fetch-parameter-provider-parameters/fetch-parameter-provider-parameters.component.ts:179]: `console.log` should be removed.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/parameter-providers/fetch-parameter-provider-parameters/fetch-parameter-provider-parameters.component.ts [hunk 3]
```diff
@@ -0,0 +1,600 @@
...
+        );
+    }
+
+    isSynced(parameterGroupConfig: ParameterGroupConfiguration): boolean {
+        return !!parameterGroupConfig.synchronized;
+    }
+
+    getParameterMappingDataSource(parameterGroupConfig: ParameterGroupConfiguration) {
+        if (!this.dataSources[parameterGroupConfig.groupName]) {
+            // const ds = new MatTableDataSource<ParameterSensitivity>();
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/parameter-providers/fetch-parameter-provider-parameters/fetch-parameter-provider-parameters.component.ts:239]: This line can be removed.

**Review by mcgilman — COMMENTED** [review]
Thanks for the PR @rfellows! Just did a quick pass of the changes and noticed some things to clean up. Still need to do a more thorough review and verification though.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/parameter-providers/parameter-providers.effects.ts [hunk 4]
```diff
@@ -61,9 +71,17 @@ export class ParameterProvidersEffects {
                             }
                         })
                     ),
-                    catchError((error) =>
-                        of(ParameterProviderActions.parameterProvidersApiError({ error: error.error }))
-                    )
+                    catchError((error: HttpErrorResponse) => {
+                        if (status === 'success') {
```
[the remark this answers is not in the document]
Comment (rfellows) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/state/parameter-providers/parameter-providers.effects.ts:75]: good catch. i think i write this before that was available. 👍

**Review by mcgilman — APPROVED** [review]
Thanks for the updates @rfellows! Will merge once CI completes.

**Pull request merged**